core/trino-spi/src/main/java/io/trino/spi/session/PropertyMetadata.java

```diff
@@ -262,12 +262,33 @@ public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(String name,
 
     public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(String name, String descriptionPrefix, Class<T> type, T defaultValue, Consumer<T> validation, boolean hidden)
     {
-        String allValues = EnumSet.allOf(type).stream()
+        return enumProperty(name, descriptionPrefix, type, EnumSet.allOf(type), defaultValue, validation, hidden);
+    }
+
+    public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(String name, String descriptionPrefix, Class<T> type, EnumSet<T> allowedValues, T defaultValue, boolean hidden)
+    {
+        return enumProperty(name, descriptionPrefix, type, allowedValues, defaultValue, value -> {}, hidden);
+    }
+
+    public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(
+            String name,
+            String descriptionPrefix,
+            Class<T> type,
+            EnumSet<T> validValues,
+            T defaultValue,
+            Consumer<T> validation,
```

Comment:
validValues and validation look like duplicates

instead of changes here, let's use a different enum class